app/uk/gov/hmrc/merchandiseinbaggagefrontend/model/core/Declaration.scala

```diff
@@ -201,7 +201,7 @@ case class DeclarationJourney(sessionId: SessionId,
         maybeExciseOrRestrictedGoods.contains(false) &&
         maybeValueWeightOfGoodsExceedsThreshold.contains(false) &&
         journeyDetailsCompleteAndDeclarationRequired &&
-        (maybeCustomsAgent.isDefined || maybeIsACustomsAgent.contains(false))
+        maybeCustomsAgent.isDefined
```

Comment:
Not sure about this

To complete a declaration you'd only need to have a defined customs agent of you answer 'yes' to the question `maybeIsACustomsAgent`.

Declaration can be complete if you answer 'No' to maybeIsACustomsAgent; if you answer 'Yes' then maybeCustomsAgent.isDefined must be true.

So it looks like there's a missing test in DeclarationSpec (my bad).

Reply:
Doh! Missed the guard on line 182